Awful.apk/src/main/java/com/ferg/awfulapp/preferences/SettingsActivity.java

```diff
@@ -302,32 +302,60 @@ protected void importFile(Intent data) {
-                                    .versionName;
-                } catch (PackageManager.NameNotFoundException e) {
-                    // rather unlikely, just show app_name without version
-                }
-                // Build the text for the About dialog
-                Resources res = getResources();
-                String aboutText = getString(R.string.about_contributors_title) + "\n\n";
-                aboutText += StringUtils.join(res.getStringArray(R.array.about_contributors_array), '\n');
-                aboutText += "\n\n" + getString(R.string.about_libraries_title) + "\n\n";
-                aboutText += StringUtils.join(res.getStringArray(R.array.about_libraries_array), '\n');
-
-                return new AlertDialog.Builder(this)
-                        .setTitle(app_version)
-                        .setMessage(aboutText)
-                        .setNeutralButton(android.R.string.ok, (dialog, which) -> {
-                        })
-                        .create();
-            default:
-                return super.onCreateDialog(dialogId);
+        if (dialogId == DIALOG_ABOUT)
+            return getAboutDialog();
+        else
+            return super.onCreateDialog(dialogId);
+    }
+
+    private Dialog getAboutDialog() {
+        return new AlertDialog.Builder(this)
+                .setTitle(getAboutDialogTitle())
+                .setMessage(getAboutDialogText())
+                .setNeutralButton(android.R.string.ok, (dialog, which) -> {
+                })
+                .create();
+    }
+
+    private String getAboutDialogTitle() {
+        String result = getText(R.string.app_name).toString();
+
+        try {
+            result += " " +
+                    getPackageManager().getPackageInfo(getPackageName(), 0).versionName;
+        } catch (PackageManager.NameNotFoundException e) {
+            // rather unlikely, just return app_name without version
         }
+
+        return result;
+    }
+
+    private String getAboutDialogText() {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        stringBuilder.append(getString(R.string.about_contributors_title)).append("\n");
+
+        for (String s : getStringArray(R.array.about_contributors_array)) {
+            stringBuilder.append("- ").append(s).append('\n');
+        }
+
+        stringBuilder.append("\n").append(getString(R.string.about_libraries_title)).append("\n");
+
+        for (String s : getStringArray(R.array.about_libraries_array)) {
+            stringBuilder.append("- ").append(s).append('\n');
+        }
+
+        stringBuilder.append('\n').append(getWebViewUserAgentString());
+
+        return stringBuilder.toString();
+    }
+
+    private String[] getStringArray(int arrayId) {
+        return getResources().getStringArray(arrayId);
+    }
+
+    private String getWebViewUserAgentString() {
+        return new WebView(this).getSettings().getUserAgentString();
```

Comment:
Actually I just realised there's a static ``WebSettings#getDefaultUserAgent`` method, we should just use that since we're not changing it anyway

Technically it would be better to have that returned by ``AwfulWebView#getUserAgentString`` so the details are encapsulated there, but if you just wanna call it in this dialog builder that's fine with me, we're not using it anywhere else anyway!